dev/validate_bom__config.py

```diff
@@ -899,6 +913,149 @@ def add_files_to_upload(self, options, file_set):
       file_set.add(options.docker_account_credentials)
 
 
+class GooglePubsubConfigurator(Configurator):
+  """Controls hal config google pub/sub configuration."""
+
+  def init_argument_parser(self, parser, defaults):
+    """Implements interface."""
+    add_parser_argument(
+        parser, 'pubsub_google_project', defaults, None,
+        help='The name of the GCE project the subscription lives in.')
+    add_parser_argument(
+        parser, 'pubsub_google_credentials_path', defaults, None,
+        help='Path to service account credentials with access to the subscription.')
+    add_parser_argument(
+        parser, 'pubsub_google_subscription_name', defaults, None,
+        help='The name of the pub/sub subscription to pull from.')
+    add_parser_argument(
+        parser, 'pubsub_google_name', defaults, None,
+        help='The logical name of the configured subscription in Spinnaker.')
+    add_parser_argument(
+        parser, 'pubsub_google_template_path', defaults, None,
+        help='Path to the Jinja message translation template.')
+
+  def validate_options(self, options):
+    """Implements interface."""
+    check_options_set(options, ['pubsub_google_project', 'pubsub_google_credentials_path', 'pubsub_google_subscription_name', 'pubsub_google_template_path'])
+    check_path_exists(options.pubsub_google_credentials_path, "pubsub_credentials_path")
+    check_path_exists(options.pubsub_google_template_path, "pubsub_template_path")
+    options.pubsub_google_enabled = options.pubsub_google_subscription_name is not None
+
+  def add_config(self, options, script):
+    """Implements interface."""
+    name = options.pubsub_google_name or None
```

Comment:
These variables arent helping you.

if not options.pubsub_google_name:
   return

subscription_cmd.append('add ' + options.pubsub_google_name)
if options.pubsub_google_project:
 subscription_cmd.append('--project ' + options.pubsub_google_project)
...
script.append('hal -q ..')
script.append(' '.join(subscription_cmd))